### DISCUSSION TITLE
Is this normal?

### DISCUSSION THREAD

**Discussion by bjethwan** [kedacore/keda#3299] [Q&A / Need Help] [answered]
### Report

At one point the target is shown as 5/2 and very next the target changes to 1667m/2...What does that mean?
```
% k get hpa
NAME                         REFERENCE         TARGETS     MINPODS   MAXPODS   REPLICAS   AGE
keda-hpa-demo-scaledobject   Deployment/demo   5/2 (avg)   1         100       1          93m

% k get hpa
NAME                         REFERENCE         TARGETS         MINPODS   MAXPODS   REPLICAS   AGE
keda-hpa-demo-scaledobject   Deployment/demo   1667m/2 (avg)   1         100       3          94m
```

### Expected Behavior

Just a display issue

### Actual Behavior

N/A

### Steps to Reproduce the Problem

1. N/A

### Logs from KEDA operator

```
example
```

### KEDA Version

2.7.1

### Kubernetes Version

1.23

### Platform

Other

**Comment by JorTurFer** [comment, marked as answer]
hey,
It's totally normal, check the unit, in the first case there isn't any unit, but in the second case there is `m` which means mili (*1000). Your HPA is using `AverageValue` as `metricType` most probably, that means that if your upstream is returning 5, between 3 instances, it's 1.666, and translated to mili `1666m`. (Reopen the issue if you still think that it's an error 😄 )

**Reply by noamgreen** [reply]
Hi , i dont think its normal 
i have same issues as i 

``` json 
  triggers:
  - metadata:
      metricName: prod-medium
      query: sum(sqs_count{name="prod-internal-processor-medium"})
      serverAddress: http://server:9090
      threshold: "100"
    type: prometheus
```


```
keda-hpa-prod-internal-tags-processor-medium-keda    Deployment/prod-internal-tags-processor-medium    4276200m/100 (avg)    1         15        15         13h
```

you can see i am in "m" and i dont understand why ? i can see "avg" but i dont understand why  

any idea ?

**Reply by JorTurFer** [reply]
It's totally normal, check the unit, in the first case there isn't any unit, but in the second case there is m which means mili (*1000). Your HPA is using AverageValue as metricType most probably, that means that if your upstream is returning `64143000`, between `15` instances, it's `4276,2`, and translated to mili `4276200m`
If you think that the value isn't correct, try doing the query directly to the prometheus server and check the differences (remember to multiply HPA value * instance count to have the raw value you get from prometheus). 
if there is a mismatch, the issue is related with prometheus scaler, but not with the `m`, `m` is just the unit for displaying the value

**Reply by noamgreen** [reply]
amm ok ... so "AverageValue as metricType most probably" can it be changed ?  i didn't see as i using "type: prometheus" 

and my "prometheus" query i can tell you i am doing "sum(sqs_count{name="prod-internal-tags-processor-medium"})"
so  the number i get is "70011" 

in the HPA 
4667400m/100 (avg) 

what i  missing is that i get from "prometheus" the absolute number  from my Q   ... and i want to scale up if the number is bigger  then 100  ... i am not sure how its scale up let say if now i have 1000 why 2 pods or 3 pods ? what is the magic calculation ....? and if the  "prometheus" give me a number why i am not seeing it in the HPA as absolute number ( as it is ). 

what am i missing ??? why i dont see the number as its in prometheus ? 
Thanks

**Reply by JorTurFer** [reply]
The value you see in the HPA is the result of the calculation, it's just how Kubernetes decided to display it. The HPA controller calculates the average of the value returned by KEDA (the raw value from your Prometheus query) between the amount of instances of your workload, this means that in this case, KEDA returns 70011, the HPA calculates the average value (you had 15 replicas) so `70011/15=4667,4`. 
As I explained above, Kubernetes doesn't use float numbers, it changes the scale multiplying the value for 1000 to change the scale to `mili` (`m`), so `4667,4*1000=4667400m`, that's why you see that value in your HPA.

You can get more information about how the calculation algorithm in the [official docs](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details). 

The scaling behavior is not a boundary to scale from it, the HPA controller will try to keep the average value closest as possible to the target value adding/removing instances, so in your scenario, you have 70000 messages and your target value is 100 messages in average, so you need `70000/100=700` but you have limited the workload up to 15, that's why you can see 4667400m in average per pod.

**Reply by tomkerkhove** [reply]
Also see https://keda.sh/docs/2.9/faq/

**Reply by gihanv** [reply]
@JorTurFer, I want to include the threshold value in milli (m) as 0/500m. Is it possible to pass this format, and if so, what are the additional parameters required?

**Reply by JorTurFer** [reply]
It depends on the trigger and KEDA version. In the scaler docs we clarify if the threshold can be float or not. 
If it can be float, you can just set 0.5, if it cant be, it's not supported

**Reply by gihanv** [reply]
@JorTurFer  values like 0.5 are not supported, It would be greatly appreciated if you could consider adding support for values like 0.5 in future versions.

**Reply by JorTurFer** [reply]
What trigger are you using and which KEDA version?

**Reply by gihanv** [reply]
Redis List and V 2.10

**Reply by JorTurFer** [reply]
Redis list doesn't support floats because list length isn't float.
Scalers docs clarify if they support floats or not, in case of supporting floats, they have a comment. e.g: prometheus
![image](https://github.com/kedacore/keda/assets/36899226/56839187-d22c-40a9-b140-ced5392d33ef)
